### ISSUE TITLE
[BUG]: https://storage.googleapis.com has been replaced by https://dl.k8s.io

### ISSUE THREAD

**Issue by mwlpeeters** [microsoft/azure-pipelines-tasks#20733]
### New issue checklist

- [x] I searched for [existing GitHub issues](https://github.com/microsoft/azure-pipelines-tasks/issues)
- [x] I read [pipeline troubleshooting guide](https://docs.microsoft.com/vsts/build-release/actions/troubleshooting)
- [x] I checked [how to collect logs](https://learn.microsoft.com/azure/devops/pipelines/troubleshooting/review-logs?view=azure-devops)

### Task name

Kubectl & KubectlInstaller

### Task version

Kubectl@1 & KubectlInstaller@0

### Issue Description

I noticed both tasks are no longer able to download the correct kubectl. In the used libraries the url https://storage.googleapis.com is used. This is changed to https://dl.k8s.io thus the tasks are now failing to download the kubectl tool.

For example:

```https://storage.googleapis.com/kubernetes-release/release/v1.30.6/bin/windows/amd64/kubectl.exe.```

Is now:

```https://dl.k8s.io/release/v1.30.6/bin/windows/amd64/kubectl.exe```

### Environment type (Please select at least one enviroment where you face this issue)

- [x] Self-Hosted
- [ ] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

Both Windows and Linux buildservers

### Relevant log output

```shell
Starting: KubectlInstaller
==============================================================================
Task         : Kubectl tool installer
Description  : Install Kubectl on agent machine
Version      : 0.247.1
Author       : Microsoft Corporation
Help         : https://aka.ms/azpipes-kubectl-installer-tsg
==============================================================================
Downloading: https://storage.googleapis.com/kubernetes-release/release/v1.30.6/bin/windows/amd64/kubectl.exe
##[error]Error: Failed to download kubectl from location https://storage.googleapis.com/kubernetes-release/release/v1.30.6/bin/windows/amd64/kubectl.exe. Error Error: Unexpected HTTP response: 404
    at Object.<anonymous> (F:\_\a\_work\_tasks\KubectlInstaller_8413c881-4959-43d5-8840-b4ea0ffc5cfd\0.247.1\node_modules\azure-pipelines-tasks-kubernetes-common\node_modules\azure-pipelines-tool-lib\tool.js:216:31)
    at Generator.next (<anonymous>)
    at fulfilled (F:\_\a\_work\_tasks\KubectlInstaller_8413c881-4959-43d5-8840-b4ea0ffc5cfd\0.247.1\node_modules\azure-pipelines-tasks-kubernetes-common\node_modules\azure-pipelines-tool-lib\tool.js:5:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  httpStatusCode: 404
}
Finishing: KubectlInstaller
```

### Full task logs with system.debug enabled

<details>
Starting: KubectlInstaller
==============================================================================
Task         : Kubectl tool installer
Description  : Install Kubectl on agent machine
Version      : 0.247.1
Author       : Microsoft Corporation
Help         : https://aka.ms/azpipes-kubectl-installer-tsg
==============================================================================
Downloading: https://storage.googleapis.com/kubernetes-release/release/v1.30.6/bin/windows/amd64/kubectl.exe
##[error]Error: Failed to download kubectl from location https://storage.googleapis.com/kubernetes-release/release/v1.30.6/bin/windows/amd64/kubectl.exe. Error Error: Unexpected HTTP response: 404
    at Object.<anonymous> (F:\_\a\_work\_tasks\KubectlInstaller_8413c881-4959-43d5-8840-b4ea0ffc5cfd\0.247.1\node_modules\azure-pipelines-tasks-kubernetes-common\node_modules\azure-pipelines-tool-lib\tool.js:216:31)
    at Generator.next (<anonymous>)
    at fulfilled (F:\_\a\_work\_tasks\KubectlInstaller_8413c881-4959-43d5-8840-b4ea0ffc5cfd\0.247.1\node_modules\azure-pipelines-tasks-kubernetes-common\node_modules\azure-pipelines-tool-lib\tool.js:5:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  httpStatusCode: 404
}
Finishing: KubectlInstaller
</details>

### Repro steps

```yml
Add to a clean pipeline

    - task: KubectlInstaller@0
      inputs:
        kubectlVersion: '$(KubernetesVersion)'
```

**Comment by Malgefor** [comment]
Running into the same issue!

**Comment by nnellanspdl** [comment]
Still an issue.

**Comment by sharebear** [comment]
Upstream fix appears to have been merged, but looks like it needs releasing, then the version updating here https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/KubernetesV1/package.json#L13

**Comment by v-bkasu** [comment]
Hi @mwlpeeters, @Malgefor, @nnellanspdl , @sharebear - could you all please confirm whether this issue is resolved or not. The changes are already deployed.

**Comment by devedse** [comment]
For us this just started happening too:
https://github.com/microsoft/azure-pipelines-tasks/issues/20888

**Comment by sharebear** [comment]
@v-bkasu we just tested now and we still got the 1.250.1 version of the Kubernetes@1 task in our pipeline, it looks like we need 1.252 for the fix to work... iirc we have no way of controlling the timing of the rollout of the task.

**Comment by devedse** [comment]
Hi v-bkasu, thanks for the fix.

I do have to admit I find it quite peculiar that it takes 2 weeks for the continuous integration for a task made for doing continuous integration to be deployed. What is the holdup that's causing this?

**Comment by devedse** [comment]
Hello @v-bkasu , did a fix release already for this? Currently we're actively waiting on this to be resolved.

**Comment by devedse** [comment]
I see a serious risk here in relying on Azure Pipeline Tasks in general if the release time for a critical fix is >2 weeks. This particular issue is a perfect example—an external URL changed, breaking functionality, and despite the fix being merged upstream, users are left waiting for an indeterminate amount of time before it actually rolls out.

If a similar issue were to occur with a dependency that cannot be easily reverted (e.g., a security vulnerability, a hard deprecation, or a domain shutdown), are we essentially saying that every Azure DevOps user will have broken pipelines for at least two weeks with no workaround? That’s an unacceptable risk for organizations that depend on CI/CD for production deployments.

CI/CD tools are supposed to enable agility, not introduce bottlenecks. What is the holdup preventing faster releases of these tasks? Is there no way to provide an emergency fast-track release process for breaking issues like this?

Would love to hear thoughts on how Microsoft plans to address this fundamental risk moving forward.

**Comment by v-bkasu** [comment]
Hi All, 

The changes have been deployed to all customers. Can you all please check and let me know.
